src/lib/logger.ts

```diff
@@ -106,6 +113,39 @@ export class Logger {
     Logger.log(event, parameters);
   }
 
+  public static logToolChange(change: ToolChangeEventType, toolId: string) {
+    let parameters: {[k: string]: any} = {
+      toolId,
+      ...change
+    };
+
+    // either flatten change.properties (for graphs) or change.data (for drawing tool),
+    // unless properties/data an array, in which case standardize on "properties" and clean up.
+    const properties = parameters.properties || parameters.data;
+    delete parameters.properties;
+    delete parameters.data;
+    if (!Array.isArray(properties)) {
+      parameters = {
+        ...parameters,
+        ...properties
+      };
+    } else {
+      // clean up MST array
+      parameters.properties = Array.from(properties);
+    }
+
+    // duck typing
+    const eventName = "operation" in change ? LogEventName.GRAPH_TOOL_CHANGE : LogEventName.DRAWING_TOOL_CHANGE;
```

Comment:
Seems like we'll need a better way of distinguishing moving forward as we log events from more tools.